frontend/rpm/template.go

```diff
@@ -293,45 +293,66 @@ func (w *specWrapper) BuildSteps() fmt.Stringer {
 
 func (w *specWrapper) PreUn() fmt.Stringer {
 	b := &strings.Builder{}
-	b.WriteString("%preun\n")
 
-	if w.Spec.Artifacts.Systemd != nil {
-		keys := dalec.SortMapKeys(w.Spec.Artifacts.Systemd.Units)
-		for _, servicePath := range keys {
-			serviceName := filepath.Base(servicePath)
-			fmt.Fprintf(b, "%%systemd_preun %s\n", serviceName)
-		}
+	if w.Spec.Artifacts.Systemd == nil {
```

Comment:
Nit: we can add a function like `w.Spec.Artifacts.Systemd.IsEmpty()`
That function can also do its own nil check.

Reply:
If `w.Spec.Artifacts.Systemd` is nil, the method `IsEmpty` can't be called on it, so it'll have to be a function and not a method.

Reply:
done

Reply:
You can call a method on a nil pointer:

https://go.dev/play/p/BoM5AGGb0Lt

Reply:
> You can call a method on a nil pointer:
> 
> https://go.dev/play/p/BoM5AGGb0Lt

TIL, thanks!